src/App.tsx

```diff
@@ -1,26 +1,336 @@
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import cN from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { ErrorNotification } from './components/ErrorNotification';
+import { filterTodosByStatus } from './utils/filterTodosByStatus';
+import { removeTodoById } from './utils/removeTodoById';
 
-const USER_ID = 0;
+const filterValues = Object.values(Filter);
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error>(Error.Default);
+  const [activeFilter, setActiveFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<number[]>([]);
+
+  const [inputText, setInputText] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [statusChangeId, setStatusChangeId] = useState<number[]>([]);
+
+  const addTodoField = useRef<HTMLInputElement>(null);
+  const errorTimeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const isAllCompleted = useMemo(() => {
+    return !todos.some(todo => todo.completed === false);
+  }, [todos]);
+
+  const hasCompleted = useMemo(() => {
+    return todos.some(todo => todo.completed === true);
+  }, [todos]);
+
+  const activeCount: number = useMemo(() => {
+    return todos.reduce((acc, todo) => {
+      if (todo.completed === false) {
+        return acc + 1;
+      }
+
+      return acc;
+    }, 0);
+  }, [todos]);
+
+  function trimTitle(text: string) {
+    return text.replace(/\s+/g, ' ').trim();
+  }
+
+  function showError(message: Error) {
+    if (errorTimeoutRef.current) {
+      clearTimeout(errorTimeoutRef.current);
+    }
+
+    setErrorMessage(message);
+    errorTimeoutRef.current = setTimeout(
+      () => setErrorMessage(Error.Default),
+      3000,
+    );
+  }
+
+  function filterToBool(filter: Filter) {
+    let boolFilter = null;
+
+    if (filter === Filter.Active) {
+      boolFilter = false;
+    } else if (filter === Filter.Completed) {
+      boolFilter = true;
+    }
+
+    return boolFilter;
+  }
+
+  function changeState(
+    id: number,
+    todosState: Dispatch<SetStateAction<Todo[]>>,
+    updatedTodo: Todo,
+  ) {
+    const filter = filterToBool(activeFilter);
+
+    todosState(prev => {
+      let changed = prev.map(todo => (todo.id === id ? updatedTodo : todo));
+
+      if (todosState === setFilteredTodos) {
+        changed = filterTodosByStatus(changed, filter);
+      }
+
+      return changed;
+    });
+  }
+
+  function handleTitleChange(newTitle: string, editingTodoId: number | null) {
+    const updateStatus = { title: newTitle };
+
+    return client
+      .patch<Todo>(`/todos/${editingTodoId}`, updateStatus)
```

Comment:
Create a function to do it. Your component shouldn't know anything about URL where you are sending the request